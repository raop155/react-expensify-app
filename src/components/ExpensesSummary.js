import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export class ExpensesSummary extends React.Component {
    render() {
        return (
            <div>
                <h1>
                    Viewing {this.props.expenseCount}
                    {this.props.expenseCount > 1 ? ' expenses ' : ' expense '}
                    totalling {numeral(this.props.expensesTotal / 100).format('$0,0.00')}
                </h1>
            </div>
        );
    }
}

const mapStateToProps = state => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
