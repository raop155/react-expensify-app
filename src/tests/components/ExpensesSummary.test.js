import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly', () => {
  const wrapper = shallow(<ExpensesSummary />);
  expect(wrapper).toMatchSnapshot();
});

test('total count with single expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={12} />);
  expect(wrapper).toMatchSnapshot();
});

test('total count with multiple expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={456} />);
  expect(wrapper).toMatchSnapshot();
});
