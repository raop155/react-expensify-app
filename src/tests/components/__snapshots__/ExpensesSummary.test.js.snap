// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
     expense 
    totalling 
    $0.00
  </h1>
</div>
`;

exports[`total count with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    2
     expenses 
    totalling 
    $4.56
  </h1>
</div>
`;

exports[`total count with single expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expense 
    totalling 
    $0.12
  </h1>
</div>
`;
